src/cachemachine/simplerepoman.py

```diff
@@ -0,0 +1,17 @@
+from typing import Any, Dict, List, Set
```

Comment:
I think the goal here is to have a class hierarchy of repository managers, but they're not organized into a class hierarchy. If you define an abstract base class that defines an interface of just `desired_images`, it will simplify some of the typing.